From a3c30babf64e8b97dc8dd5e930fd839c92937946 Mon Sep 17 00:00:00 2001
From: Ofir Fedida <ofedida@marvell.com>
Date: Wed, 19 Jul 2017 14:39:53 +0300
Subject: [PATCH 1094/1345] a80x0: add support for Armada 8k OCP board

commit  5745c4376e6c802a1d35221eb85a522f905455a9 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

this patch introduces support for new Armada 8k OCP board.
- added new device tree (armada-8040-ocp)

verified interfaces:
2 * 10G ethernet ports
SPI
MMC
UART

Change-Id: I00c9412c88c9b3c09fe5b3d0296d994597ba2cb0
Signed-off-by: Ofir Fedida <ofedida@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/41849
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../bindings/arm/marvell/armada-7k-8k.txt          |    3 +
 arch/arm64/boot/dts/marvell/Makefile               |    1 +
 arch/arm64/boot/dts/marvell/armada-8040-ocp.dts    |  151 ++++++++++++++++++++
 3 files changed, 155 insertions(+)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-8040-ocp.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
index 3bbaa98..51af1bd 100644
--- a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
@@ -71,6 +71,9 @@ Boards:
    - "marvell,armada8040-mcbin"
       when the board being used is the MACCHIATOBin community board
 
+   - "marvell,armada8040-ocp"
+      when the board being used is the OCP board
+
 Example:
 
 compatible = "marvell,armada7040-db-router", "marvell,armada7040-db-nand", "marvell,armada7040-db-all-pcie", "marvell,armada7040-pd",
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 6fa5915..e918eda 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -31,6 +31,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-C.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-D.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-E.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-ocp.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-80x0-customer.dtb
 
 always		:= $(dtb-y)
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts b/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts
new file mode 100644
index 0000000..5bbc4af
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2017 Marvell Technology Group Ltd.
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPLv2 or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This library is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This library is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Device Tree file for OCP Armada 8040 board platform
+ */
+
+#include "armada-8040.dtsi"
+
+/ {
+	model = "Armada 8040 OCP";
+	compatible = "marvell,armada8040-ocp", "marvell,armada8040",
+			"marvell,armada-ap806-quad", "marvell,armada-ap806";
+
+	memory@00000000 {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x80000000>;
+	};
+
+	ap806 {
+		config-space {
+			serial@512000 {
+				status = "okay";
+			};
+			sdhci@6e0000 {
+				vqmmc-supply = <&vccq_mmc_reg>;
+				bus-width = <8>;
+				no-1-8-v;
+				non-removable;
+				marvell,xenon-emmc;
+				status = "okay";
+			};
+			vccq_mmc_reg: regulator@0 {
+				compatible = "regulator-fixed";
+				regulator-name = "vccq-mmc-reg";
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+				status = "okay";
+			};
+		};
+	};
+
+	cpn-110-master {
+		config-space {
+			comphy0: comphy {
+				status = "okay";
+			};
+			i2c@701000 {
+				clock-frequency = <100000>;
+				pinctrl-names = "default";
+				pinctrl-0 = <&cpm_i2c0_pins>;
+				status = "disabled";
+			};
+			gop {
+				emac0_0: mac0 {
+					phys = <&comphy0 4 COMPHY_SFI>;
+					phy-names = "comphy";
+					phy-mode = "sfi"; /* lane-4 */
+				};
+				emac2_0: mac2 {
+					phy-mode = "none";
+				};
+				emac3_0: mac3 {
+					phy-mode = "none";
+				};
+			};
+			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
+				pinctrl-names = "default";
+				eth0_0: eth0@010000 {
+					status = "okay";
+				};
+			};
+		};
+	};
+
+	cpn-110-slave {
+		config-space {
+			comphy1: comphy {
+				status = "okay";
+			};
+			gop {
+				emac0_1: mac0 {
+					phys = <&comphy0 4 COMPHY_SFI>;
+					phy-names = "comphy";
+					phy-mode = "sfi"; /* lane-4 */
+				};
+				emac2_1: mac2 {
+					phy-mode = "none";
+				};
+				emac3_1: mac3 {
+					phy-mode = "none";
+				};
+			};
+			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
+				pinctrl-names = "default";
+				eth0_1: eth0@010000 {
+					status = "okay";
+				};
+				eth1_1: eth1@020000 {
+					status = "disabled";
+				};
+				eth2_1: eth2@030000 {
+					status = "disabled";
+				};
+			};
+		};
+	};
+};
+
-- 
1.7.9.5

